### PR TITLE
feat(exporting): allow passing writer_opts to PyArrow ParquetWriter and CSVWriter

### PR DESCRIPTION
**Pull request by jayceslesar**
Small tweak to allow passing of writer specific args/kwargs. Also fixes a typo

### CODE REVIEW

File: ibis/backends/base/__init__.py [hunk 1]
```diff
@@ -546,8 +546,9 @@ def to_parquet(
         self._import_pyarrow()
         import pyarrow.parquet as pq
 
+        writer_opts = kwargs.get("writer_opts", {})
         with expr.to_pyarrow_batches(params=params) as batch_reader:
-            with pq.ParquetWriter(path, batch_reader.schema) as writer:
+            with pq.ParquetWriter(path, batch_reader.schema, **writer_opts) as writer:
```
Comment (cpcloud) [inline ibis/backends/base/__init__.py:551]: How about instead passing `**kwargs`?

File: ibis/backends/base/__init__.py [hunk 2]
```diff
@@ -581,8 +582,9 @@ def to_csv(
         self._import_pyarrow()
         import pyarrow.csv as pcsv
 
+        writer_opts = kwargs.get("writer_opts", {})
         with expr.to_pyarrow_batches(params=params) as batch_reader:
-            with pcsv.CSVWriter(path, batch_reader.schema) as writer:
+            with pcsv.CSVWriter(path, batch_reader.schema, **writer_opts) as writer:
```
Comment (cpcloud) [inline ibis/backends/base/__init__.py:587]: Same as the Parquet version: How about instead passing `**kwargs`?

**Review by cpcloud — CHANGES_REQUESTED** [review]
Thanks for PR!

Can you add a test, one for each of parquet and csv code paths? Putting them in `ibis/backends/tests/test_export.py` is fine.

File: ibis/backends/base/__init__.py (hunk 1 above)
Comment (jayceslesar) [inline ibis/backends/base/__init__.py:551]:
```suggestion
            with pq.ParquetWriter(path, batch_reader.schema, **kwargs) as writer:
```

File: ibis/backends/base/__init__.py (hunk 2 above)
Comment (jayceslesar) [inline ibis/backends/base/__init__.py:587]:
```suggestion
            with pcsv.CSVWriter(path, batch_reader.schema, **kwargs) as writer:
```

**Comment by jayceslesar** [comment]
Working on adding tests but have had to exclude duckdb as its export options are not the same as pyarrow https://duckdb.org/docs/sql/statements/copy#csv-options

**Comment by jayceslesar** [comment]
Heads up this is failing because the duckdb backend doesnt use the writers it just uses the duckdb copy kwargs, so I think we should exclude duckdb from this test unless we wanted it to dump normally, and then iterate the output in batches and stream to the parquet writer 🤷

**Comment by cpcloud** [comment]
I opted for excluding duckdb from the tests you added.

Thank you for address my feedback!

**Review by cpcloud — APPROVED** [review]

**Pull request merged**